pkg/util/cmapichecker/cmapichecker.go

```diff
@@ -0,0 +1,166 @@
+		},
+	}, nil
+}
+
+func (o *cmapiChecker) Client() (client.Client, error) {
+	if o.client != nil {
+		return o.client, nil
+	}
+
+	cl, err := o.clientBuilder()
+	if err != nil {
+		return nil, err
+	}
+	o.client = cl
+
+	return o.client, nil
+}
+
+// Check attempts to perform a dry-run create of a cert-manager *v1alpha2*
+// Certificate resource in order to verify that CRDs are installed and all the
+// required webhooks are reachable by the K8S API server.
+// We use v1alpha2 API to ensure that the API server has also connected to the
+// cert-manager conversion webhook.
+func (o *cmapiChecker) Check(ctx context.Context) *ApiCheckError {
+	cert := &cmapi.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "cmapichecker-",
+		},
+		Spec: cmapi.CertificateSpec{
+			DNSNames:   []string{"cmapichecker.example"},
+			SecretName: "cmapichecker",
+			IssuerRef: cmmeta.ObjectReference{
+				Name: "cmapichecker",
+			},
+		},
+	}
+
+	// while creating client: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
+	cl, err := o.Client()
+	if err != nil {
+		return &ApiCheckError{
+			SimpleError:     ErrAPIServerUnreachable,
+			UnderlyingError: err,
+		}
+	}
+
+	// error finding the scope of the object: failed to get restmapping: no matches for kind "Certificate" in group "cert-manager.io"
+	// Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": service "cert-manager-webhook" not found
+	// Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp 10.96.38.90:443: connect: connection refused
+	// Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": x509: certificate signed by unknown authority (possibly because of "x509: ECDSA verification failure" while trying to verify candidate authority certificate "cert-manager-webhook-ca")
+	// conversion webhook for cert-manager.io/v1alpha2, Kind=Certificate failed: Post "https://cert-manager-webhook.cert-manager.svc:443/convert?timeout=30s": x509: certificate signed by unknown authority
+	if err := cl.Create(ctx, cert); err != nil {
+		return &ApiCheckError{
+			SimpleError:     translateToSimpleError(err),
+			UnderlyingError: err,
+		}
+	}
+	return nil
+}
+
+func translateToSimpleError(err error) error {
+	s := err.Error()
+
+	if regexErrCertManagerCRDsNotFound.MatchString(s) {
+		return ErrCertManagerCRDsNotFound
+	} else if regexErrCertManagerAPIEndpointsNotEstablished.MatchString(s) {
+		return ErrCertManagerAPIEndpointsNotEstablished
+	} else if regexErrWebhookConnectionFailure.MatchString(s) {
+		return ErrWebhookConnectionFailure
+	} else if regexErrWebhookCertificateFailure.MatchString(s) {
+		return ErrWebhookCertificateFailure
+	}
+
+	return err
```

Comment:
This will result in duplicate error messages printed to the console when the `--verbose` flag is supplied and the error does not match any of the regexps

Reply:
Indeed, I'll fix that